OSDOCS-17654: Adding customer global pull secret in HCP for ROSA (#105229)
openshift/openshift-docs: modules/images-update-global-pull-secret.adoc

= Updating the global cluster pull secret

[role="_abstract"]

> To add new registries or update authentication for your {product-title} cluster, you can update the global pull secret by appending new credentials to the _additional-pull-secret_. In order to do that, you can use the `oc set data secret/additional-pull-secret -n kube-system` command. Hypershift will take care of the new credential propagation among the HostedCluster nodes.

Reviewer: ```suggestion
To add new registries or update authentication for your {product-title} cluster, you can update the global pull secret by appending new credentials to the _additional-pull-secret_. To do this, you can use the `oc set data secret/additional-pull-secret -n kube-system` command. Hypershift manages the new credential propagation among the HostedCluster nodes.
```

> This feature provides a dedicated mechanism to separate your private credentials from the pull secret managed by the service, ensuring cluster functionality while restricting external visibility of your sensitive data. This separation allows you to independently rotate secrets and maintain exclusive ownership for compliance without impacting core managed service operations.

Author: conditona;ized it for HCP

> Managed services already have some immutable entries in the file, and you will not be able to modify those. If you are in this situation, you can follow this approach to use the same registry with different credentials.

Author: added condition for HCP

`<pull_secret_location>`:: Specifies the path to the new pull secret file.
--
+
This merges your additional pull secret with the original HostedCluster pull secret, making it available to all nodes in the cluster.

. Optional: Modify the additional pull secret added by entering the following command:
+
[source,terminal]
----
$ oc edit secret <additional-pull-secret> -n kube-system
----
+
This will result in the following states of the each pull secret:

> * *Original*: inmutable

Reviewer 2: ```suggestion
* *Original*: immutable
```
